newIDE/app/src/Utils/GDevelopServices/Project.spec.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import {
+  extractFilenameFromProjectResourceUrl,
+  extractProjectUuidFromProjetResourceUrl,
+} from './Project';
+
+describe('Project service', () => {
+  describe('extractFilenameFromProjectResourceUrl', () => {
+    it('extracts filename from a gdevelop.io bucket URL', () => {
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy'
+        )
+      ).toBe('Pea-Happy');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png'
+        )
+      ).toBe('Pea-Happy.png');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy?some=parameter'
+        )
+      ).toBe('Pea-Happy');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png?some=parameter'
+        )
+      ).toBe('Pea-Happy.png');
+    });
+    it('extracts filename from a gdevelop.io bucket URL, and handles sub folders', () => {
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy'
+        )
+      ).toBe('Pea-Happy');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png'
+        )
+      ).toBe('Pea-Happy.png');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy?some=parameter'
+        )
+      ).toBe('Pea-Happy');
+      expect(
+        extractFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/a9fe5bce-de39-4147-a669-93fc5cd69632/resources/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png?some=parameter'
+        )
+      ).toBe('Pea-Happy.png');
+    });
+    it('extracts filename from a classic URL', () => {
```

Comment:
Same as below. Plus I don't know why the expected value contains the query parameters and the hash